### PR TITLE
Surgery - Add Additional Settings to Allow Patient to Remain Awake During Surgeries

### PR DESCRIPTION
**Pull request by Belhun**
**When merged this pull request will:**

- Added 1 Dropdown Setting with 4 options In the Surgery category.(In XEH_preInit)
  - Option 1 Surgery Causes Unconsciousness: An incision causes cardiac arrest if the patient is not unconscious and sedated(With Etomidate). Uses a Switch Statement in fnc_incisionLocal.sqf 
    - Checks if the patient Has Etomidate and if they are in an unconscious state. 
  - Option 2 Unconsciousness Required for Surgery: Surgery can only begin if the patient is unconscious and sedated(With Etomidate).
    - Changed the First if statement in fnc_incisionLocal from checking for uncon_requieredForAction to Checking for unconSurgery_requieredForAction equals 1. This statement checks to make sure the patient is unconscious and if they aren't then Surgery is aborted(This setting is the old uncon_requieredForAction, Which has been brought into the switch state due to the similarity of the job)
    - Removed Setting "uncon_requieredForAction" and the localizations for the same setting 
  - Option 3 No Unconsciousness from Surgery: The patient is able to stay awake through the entirety of the surgery, experiencing a little pain and a slight increase in heart rate (Etomidate not required). Uses a Switch Statement in fnc_incisionLocal.sqf 
    - This Case Uses the adjustPainLevel and addMedicationAdjustment, to cause pain and increase in heart rate[There is a chance this can be simplified with just the addMedicationAdjustment, but was unsure when searching for similar use cases in Code so went with 2 separate functions]
  - Option 4 Surgery Anesthesia: The patient can remain awake for the duration of the surgery. However, if Etomidate hasn't been applied, the patient will lose consciousness and go into CA. There are two cases in the switch statement in fnc_incisionLocal.sqf. 
    - The first case checks the unconSurgery_requieredForAction setting and whether there's no Etomidate in the patient's system. If so, it induces cardiac arrest by causing a heart rate increase of 200 and forces the patient into unconsciousness. 
    - The second case applies when there's Etomidate in the patient's system. In this scenario, the incision can proceed without complications, allowing the surgery to be carried out without issues.
    
- Added English Translations For All new Text Used in Settings Area

### IMPORTANT

- [x] [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- [x] Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

File: addons/surgery/XEH_preInit.sqf [hunk 1]
```diff
@@ -169,13 +169,16 @@ PREP_RECOMPILE_END;
...
+//1Unconsciouness Required for Surgery: Pationt must be Unconsciouness to do Surgery
+//2No Unconsciouness from Surgery: They will not be knocked Unconsciouness when an insision in performed
+//3Surgery Anesthesia: The patient is able to stay awake through the entirety of the surgery. If Etomidate has not been applied, then the patient will lose consciousness.
 [
-    QGVAR(uncon_requieredForAction),
-    "CHECKBOX",
-    [LLSTRING(uncon_requieredForActions), LLSTRING(uncon_requieredForActions_DESC)],
+    QGVAR(unconSurgery_requieredForAction),
+    "LIST",
+    LLSTRING(unconSurgery_requieredForAction),
```
Comment (BlueTheKing) [inline addons/surgery/XEH_preInit.sqf]: Add description to explain each of the settings

File: addons/surgery/functions/fnc_incisionLocal.sqf [hunk 2]
```diff
@@ -19,7 +19,7 @@
 
 params ["_medic", "_patient", "_bodyPart"];
 
-if (GVAR(uncon_requieredForAction)) then {
+if (GVAR(unconSurgery_requieredForAction) == 1) then {
     if !(IS_UNCONSCIOUS(_patient)) exitWith {
```
Comment (BlueTheKing) [inline addons/surgery/functions/fnc_incisionLocal.sqf]: Put them in one line
```suggestion
if (GVAR(unconSurgery_requieredForAction) == 1 && !(IS_UNCONSCIOUS(_patient))) exitWith {
```

File: addons/surgery/functions/fnc_incisionLocal.sqf [hunk 3]
```diff
@@ -53,10 +53,24 @@ _patient setVariable [QGVAR(fractures), _fractureArray, true];
...
+            [_patient, "Pain", 10, 40, 200, 0, 40] call ACEFUNC(medical_status,addMedicationAdjustment);
+            [_target, true] call ACEFUNC(medical,setUnconscious);
+        };
+        case (GVAR(unconSurgery_requieredForAction) == 3): {};
+        case ((_count == 0 || !(IS_UNCONSCIOUS(_patient))) && (GVAR(unconSurgery_requieredForAction) == 0)): 
+        {
+            [_patient, "Pain", 10, 40, 200, 0, 40] call ACEFUNC(medical_status,addMedicationAdjustment);
+            [_target, true] call ACEFUNC(medical,setUnconscious);
+        };
     };
```
Comment (BlueTheKing) [inline addons/surgery/functions/fnc_incisionLocal.sqf]: No need for switch statement
```suggestion
    if ((GVAR(unconSurgery_requieredForAction) == 0 && !(IS_UNCONSCIOUS(_patient)) && _count == 0) || (GVAR(unconSurgery_requieredForAction) == 3 && _count == 0)) exitWith {
        [_patient, "Pain", 10, 40, 200, 0, 40] call ACEFUNC(medical_status,addMedicationAdjustment);
        [_target, true] call ACEFUNC(medical,setUnconscious);
    };

    if (GVAR(unconSurgery_requieredForAction) == 2 && _count == 0) then {
        [_patient, 0.4] call ACEFUNC(medical_status,adjustPainLevel);
        [_patient, "Pain", 10, 40, 30, 0, 40] call ACEFUNC(medical_status,addMedicationAdjustment);
    };
```

File: addons/surgery/stringtable.xml [hunk 4]
```diff
@@ -1105,5 +1079,21 @@
...
+        <Key ID="STR_KAT_Surgery_Causes_Unconsciouness">
+            <English>Surgery Causes Unconsciouness</English>
+        </Key>
+        <Key ID="STR_KAT_Surgery_Unconsciouness_Required">
+            <English>Unconsciouness Required for Surgery</English>
+        </Key>
+        <Key ID="STR_KAT_Surgery_No_Unconsciouness">
+            <English>No Unconsciouness from Surgery</English>
+        </Key>
+        </Key>
```
Comment (BlueTheKing) [inline addons/surgery/stringtable.xml:1096]: Remove extra closing tag
```suggestion
```

File: addons/surgery/functions/fnc_openReductionProgressLocal.sqf [hunk 5]
```diff
@@ -20,7 +20,7 @@
 
 params ["_medic", "_patient", "_bodyPart", "_entry"];
 
-if (GVAR(uncon_requieredForAction)) then {
+if (GVAR(unconSurgery_requieredForAction == 1)) then {
     if !(IS_UNCONSCIOUS(_patient)) exitWith {
```
Comment (BlueTheKing) [inline addons/surgery/functions/fnc_openReductionProgressLocal.sqf]:
```suggestion
if (GVAR(unconSurgery_requieredForAction) == 1 && !(IS_UNCONSCIOUS(_patient))) exitWith {
```

**Review by BlueTheKing — CHANGES_REQUESTED** [review]

File: addons/surgery/functions/fnc_incisionLocal.sqf (hunk 3 above)
Comment (Belhun) [reply addons/surgery/functions/fnc_incisionLocal.sqf:74]: That makes a lot more sense. Though, would it make more sense to check if the Etomidate count is zero before checking if the other states are true? I'm new to this, to be honest, so please let me know if my logic is flawed.

Comment (BlueTheKing) [reply addons/surgery/functions/fnc_incisionLocal.sqf]: Yeah you could, don't think it'd make too much of a difference in this case.

Comment (Belhun) [reply addons/surgery/functions/fnc_incisionLocal.sqf]: Looking back it and talking with my friend, realized that it was better with the way you suggested

**Comment by Belhun** [comment]
Fixed Requested Changes and merged Spanish Translations into my Forked version

**Comment by Belhun** [comment]
Oh also, on a side note, i was able to use hemtt and dev and launch and test to make sure it worked(which these settings do work), but I couldn't Build it/release it with hemtt(it said it is a hemtt issue, an not something i did but still thought Id say something )

**Review by BlueTheKing — APPROVED** [review]
Builds, tested, looks good.

**Review by MiszczuZPolski — APPROVED** [review]

**Pull request merged**